src/Application/Service/ConfigurationInterface.php

```diff
@@ -15,5 +15,12 @@ interface ConfigurationInterface
      */
     public function get(string $key, $default = null);
 
+    /**
+     * Set a parameter based on a key.
+     *
+     * @param mixed $value
+     */
+    public function set(string $key, $value): void;
```

Comment:
Personally I don't like making Config object mutable, mutable global state is evil and should be avoided. What you think about `withNewEntry` method? It will `clone` object and returns modified instance, but keeping original object untouched.